invenio_app_ils/ui/src/invenio_app_ils/pages/frontsite/Documents/DocumentsDetails/DocumentMetadata/components/DocumentStats.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from 'react';
+import { Icon, Table, Message } from 'semantic-ui-react';
+import { Error } from '../../../../../../common/components';
+import { http } from '../../../../../../common/api/base';
+
+export class DocumentStats extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: { views: { count: '-', unique_count: '-' } },
+      error: {},
+    };
+  }
+
+  componentDidMount() {
+    this.fetchStats();
+  }
+
+  fetchStats = async () => {
+    const { document } = this.props;
+    const request = {
+      views: {
+        stat: 'record-view',
+        params: {
+          pid_value: document.pid,
+        },
+      },
+    };
+    try {
+      const response = await http.post('/stats', request);
+      this.setState({ data: response.data });
+    } catch (error) {
+      this.setState({ error: error });
+    }
+  };
+
+  render() {
+    const { document } = this.props;
+    const { data, error } = this.state;
+    const hasItems = document.metadata.items.hits.length || false;
+    const hasEitems = document.metadata.eitems.hits.length || false;
+    return (
+      <>
+        {(hasItems || hasEitems) && (
```

Comment:
what if it had items and eitems but they were deleted? also those are views for the document so i'm not sure why do you base it on existence of items but maybe I'm missing something

Reply:
I was looking for a way to show/hide the downloads thing and got carried away, I will make the download stats available only if there are eitems present.